File: apps/api/app/apis/dtos/worksite.py
from enum import Enum
from pydantic import BaseModel, ConfigDict
from typing import Optional


class Status(str, Enum):
    ACTIVE = "Active"
    INACTIVE = "Inactive"


class WorkSiteCreateSchema(BaseModel):
    name: str
    address: str
    city: str
    state: str
    zip_code: str
    contact_person: str
    contact_phone: str
    status: Status


class WorkSiteEditSchema(BaseModel):
    id: int
    name: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    zip_code: Optional[str] = None
    contact_person: Optional[str] = None
    contact_phone: Optional[str] = None
    status: Optional[str] = None


class WorkSiteResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    address: str
    city: str
    state: str
    zip_code: str
    contact_person: str
    contact_phone: str
    status: str
    organization_id: int

File: apps/web/src/types/worksite.ts
export type WorkSite = {
  id?: number; // Optional because it's not required when creating a new work site
  name: string; // Name of the work site
  address: string; // Address of the work site
  city: string; // City where the work site is located
  state: string; // State where the work site is located
  zip_code: string; // Zip code of the work site
  contact_person: string; // Name of the contact person for the work site
  contact_phone: string; // Phone number of the contact person
  status: "Active" | "Inactive"; // Status of the work site (Active or Inactive)
};
